staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -331,9 +370,16 @@ func (s *APIAggregator) AddAPIService(apiService *v1.APIService) error {
 	}
 	proxyHandler.updateAPIService(apiService)
 	if s.openAPIAggregationController != nil {
-		s.openAPIAggregationController.AddAPIService(proxyHandler, apiService)
+		// this is calling a controller.  It should already handle being async.
+		go func() {
```

Comment:
this is suspicious. the `AddAPIService` is merely pushing tasks into queue, we don't need to spawn a new routine, is it?